craft_parts/packages/apt_cache.py

```diff
@@ -43,6 +43,36 @@
 _HASHSUM_MISMATCH_PATTERN = re.compile(r"(E:Failed to fetch.+Hash Sum mismatch)+")
 
 
+class LogProgress(apt.progress.base.AcquireProgress):
+    """Internal Base class for text progress classes."""
+
+    def __init__(self):
+        self._id = 1
+
+    def fail(self, item: apt.apt_pkg.AcquireItemDesc) -> None:
+        """Handle failed item."""
+        apt.progress.base.AcquireProgress.fail(self, item)
+        if item.owner.status == item.owner.STAT_DONE:
+            logger.debug("Ign %s", item.description)
+        else:
+            logger.error("Err %s", item.description)
```

Comment:
Consider logging it all to debug and use the exception as the interface to provide the error message back to the app. Otherwise may get a little tricky to filter.